### 1. Evauate Chile data

In [1]:
import sys
sys.path.append("../../")
sys.path.append("../../../")
from tasks.data_loader.src.utils import *
import json

In [2]:
file = load_file("../input/Chile_V1_210108.json")

In [3]:
sent_label_map = labeled_sentences_from_dataset(file)

In [4]:
len(sent_label_map)

188046

In [18]:
sent_label_map['0002998_1']

{'text': 'Tipo Norma :Acuerdo 51', 'labels': []}

### Sentences that start with a comma

In [34]:
starting_comma = [(sent_id, sent) for sent_id, sent in sent_label_map.items() if len(sent['text'].strip()) != 0 and sent['text'][0] == ","]
len(starting_comma), starting_comma[20:30]

(1201,
 [('0058c31_89',
   {'text': ', tendra una validez de dos años a partir de la entrada en vigencia de la presente resolucion y sera prorrogado automaticamente en la medida que no se produzcan cambios a nivel operativo en los procedimientos autorizados por este Servicio o a nivel territorial, que pudieran afectar el estatus fitosanitario del material que sera exportado a Chile',
    'labels': []}),
  ('0066d2d_36', {'text': ', Wodyetia bifurcata y Yucca spp', 'labels': []}),
  ('00a4528_12',
   {'text': ', de fecha 31 de diciembre de 2018, Fija tarifa por prestacion de servicios de contencion inicial de contaminacion en el territorio antartico',
    'labels': []}),
  ('00abdb9_76',
   {'text': ', en el territorio que sera compartido', 'labels': []}),
  ('01203a9_23',
   {'text': ', representada por el señor Agustin de La Fuente Pedreros, con fecha 23 de julio de 2001',
    'labels': []}),
  ('01203a9_82',
   {'text': ', de lunes a jueves, y de 8:30 a 16:30 hrs', 'labels': []}),
  

### Sentences that are empty

In [44]:
empty = [(sent_id, sent) for sent_id, sent in sent_label_map.items() if len(sent['text'].strip()) == 0]
len(empty), empty[20:30]

(1324,
 [('04c353f_11', {'text': '', 'labels': []}),
  ('0515f4a_11', {'text': '', 'labels': []}),
  ('052b9ad_44', {'text': '', 'labels': []}),
  ('056c8c3_38', {'text': '', 'labels': []}),
  ('0584db4_19', {'text': '', 'labels': []}),
  ('058a53c_16', {'text': '', 'labels': []}),
  ('0594332_13', {'text': '', 'labels': []}),
  ('05cd990_24', {'text': '', 'labels': []}),
  ('05d75c9_19', {'text': '', 'labels': []}),
  ('05e9e34_19', {'text': '', 'labels': []})])

### Sentences that start with a parenthesis

In [45]:
starting_parenthesis = [(sent_id, sent) for sent_id, sent in sent_label_map.items() if len(sent['text'].strip()) != 0 and sent['text'][0] == ")"]
len(starting_parenthesis), starting_parenthesis[20:30]

(642,
 [('1575906_230',
   {'text': '), de 2011, del Ministerio de Vivienda y Urbanismo, dando una respuesta adecuada a las particularidades territoriales, condiciones de terreno y a los diferentes tipos de familias, en el marco de una politica integral que priorice la calidad, la equidad y la integracion social',
    'labels': []}),
  ('1575906_232',
   {'text': '), de 2014, correspondiente a Subsidio Habitacional para Proyectos de Integracion Social, en especifico lo que establece el punto 8.1',
    'labels': []}),
  ('15c8f2b_19',
   {'text': '), duraznero (Prunus persica) y nectarino (Prunus persica var nucipersica), procedentes de Australia',
    'labels': []}),
  ('15c8f2b_23', {'text': '), en la letra a', 'labels': []}),
  ('16c1d7d_50', {'text': ') y yuyo (Brassica rapa syn', 'labels': []}),
  ('17c4c3f_19',
   {'text': '), la cual se regira por las disposiciones contenidas en la Norma General de Certificacion de Semillas, complementadas por la presente norma especifica',
    '

### Sentences that start with a number

In [46]:
starting_num = [(sent_id, sent) for sent_id, sent in sent_label_map.items() if len(sent['text'].strip()) != 0 and sent['text'][0].isnumeric()]
len(starting_num), starting_num[1], starting_num[20:30]

(6170,
 ('001d742_18',
  {'text': '2º.- Fijase en el equivalente a 6.750 Unidades de Fomento (U.) el monto de los recursos que se destinaran al financiamiento del subsidio directo en el llamado extraordinario dispuesto en el numero anterior. Dicho monto se distribuira por tipo de postulacion (individual o colectiva), atendiendo al porcentaje de postulantes que se inscriban durante el "periodo de postulacion", segun tipo de postulacion, en relacion al total de postulantes inscritos, alcanzando el numero de postulantes seleccionados al numero de Certificados de Subsidio susceptibles de financiar con los recursos previstos para el otorgamiento del subsidio directo.',
   'labels': []}),
 [('00abdb9_32', {'text': '12 634 5459 Fresia 4100-7315', 'labels': []}),
  ('00abdb9_33', {'text': '13 619 5459 Collihuinco 4100-7330', 'labels': []}),
  ('00abdb9_34', {'text': '14 610 5453 Collihuinco 4100-7330', 'labels': []}),
  ('00abdb9_35', {'text': '15 610 5447 Collihuinco 4100-7330', 'labels': []}

### Sentences that end in a connector like "de", "un", "y", etc (also repeated sentences?)

In [65]:
connectors = {"de", "un", "y", "que", "el", "la", "los", "lo", "las", 
              "les", "ellos", "ellas", "por", "cual", "una", "unas", "unos",
             "en", "es", "esta"}
ending_connector = [(sent_id, sent) for sent_id, sent in sent_label_map.items() if len(sent['text'].strip()) != 0 and sent['text'].split()[-1] in connectors]
len(ending_connector), ending_connector[10:30]

(2392,
 [('013a174_36',
   {'text': '(iii) Permitir el desarrollo de capacidades exportadoras que faciliten la realizacion de estrategias sustentables en los mercados externos, en el segmento de las pequeñas y',
    'labels': []}),
  ('013a174_41',
   {'text': '(ii) Generacion y difusion de informacion, para que las empresas silvoagropecuarias chilenas tengan acceso a informacion y conocimientos pertinentes y de alta calidad, sobre las caracteristicas y condiciones de los mercados, clientes y consumidores extranjeros, y',
    'labels': []}),
  ('01a45df_29',
   {'text': 'Numero de Grupo de Trabajo Minimo por', 'labels': []}),
  ('01a45df_30',
   {'text': 'Familias con Prestador de Servicios de', 'labels': []}),
  ('01a45df_33', {'text': 'tiempo completo del area de la', 'labels': []}),
  ('01a45df_35', {'text': 'legal, quien estara a cargo de la', 'labels': []}),
  ('01a45df_37', {'text': 'mas un profesional del area de la', 'labels': []}),
  ('01a45df_42', {'text': 'tiempo completo de

# Results from data checking 2: Chile_V1_210108.json
- Total sents: 188046
- Empty sentences: 1324
- Sentences that start with a comma: 1201
- Sentences that start with a ")": 642
- Sentences that start with a number: 6170 (Could also include good examples, like bullet points!)
- Sentences being cut: ~2392, tried checking for connectors ("de", "un", "y", "que", etc.) at the end of sentence and also commas 

# Results from data checking 1: Chile.json
- 154436 sentences instead of 79953
- Sentences that start with a comma: 1018
    - ", en el punto 4,3"
    - ", de 10 de julio de 2018;"
    - ", Empresa Eléctrica de Arica S"
    - ", y su aplicación será competencia del Juzgado de Policía Local de la comuna"
    
- Sentences that start with a ")": 1338
    - "), de fecha 19 de diciembre de 2008, y sus modificaciones, que autoriza llamados a postulación para subsidios habitacionales en sistemas y programas habitacionales que indica durante el año 2009 y señala el monto de los recursos destinados, entre otros, a la atención a través del sistema regulado por el DS Nº 145 (V"
    - "), en la letra a"
    
- Sentences that start with a number: 12389
    - Bad examples:
        - "25 634 5414 Los Muermos 4115-7315"
        - "3.815 de 2003"
        - "11 Aguas Calientes 4 636365 7353860 8 y 10"
    - Unclear examples:
        -  "2018 del Director (S) del Servicio de Salud Viña del Mar - Quillota"
    - Good examples:
        - "2º Establécese el trámite: TCB1 "Declaración de Plantas de Biogás" para los siguientes tipos de Plantas:"
        - "1) Que, el artículo 72º-19 de la Ley establece que la Comisión deberá fijar, mediante resolución exenta, las normas técnicas que rijan los aspectos económicos, de seguridad, coordinación, calidad, información y económicos del funcionamiento del sector eléctrico, debiendo establecer un plan de trabajo anual que permita proponer, facilitar y coordinar el desarrollo de tales normas, en el marco de un proceso público y participativo, cuyas normas deben ser establecidas en un reglamento;"
        - "6,5 La madera de los embalajes y los pallets deberá estar libre de corteza y de daños causados por insectos"
        

- Sentences being cut: ~8205, tried checking for connectors ("de", "un", "y", "que", etc.) at the end of sentence and also commas 
    - "+ Se lavará el carro, en el área sucia y"
    - "+ Debido al riesgo de los pacientes que en él"
    - "más dos profesionales del área de la"
    - "construcción, un profesional del"